plugins/modules/storage/zfs/zfs.py

```diff
@@ -194,19 +194,25 @@ def set_property(self, prop, value):
             self.module.fail_json(msg=err)
 
     def set_properties_if_changed(self):
+        diff = {'before': {}, 'after': {}}
         current_properties = self.get_current_properties()
         for prop, value in self.properties.items():
-            if current_properties.get(prop, None) != value:
+            current_value = current_properties.get(prop, None)
+            if current_value != value:
                 self.set_property(prop, value)
+                diff['before'][prop] = current_value
+                diff['after'][prop] = value
         if self.module.check_mode:
-            return
+            return diff
         updated_properties = self.get_current_properties()
-        for prop in self.properties:
+        for prop in diff['before']:
             value = updated_properties.get(prop, None)
             if value is None:
                 self.module.fail_json(msg="zfsprop was not present after being successfully set: %s" % prop)
-            if current_properties.get(prop, None) != value:
+            if diff['before'][prop] != value:
```

Comment:
Same comment here: `diff['before']` is populated from the keys of `self.properties` and values from `current_properties`, so I think using `diff['before']` just makes the code a bit more confusing.

Reply:
I also reverted back this code line.